articles/protocols/oauth2/index.md

```diff
@@ -66,13 +64,117 @@ We will now have a more detailed look on how the protocol works. As we will see
 
 ## Authorization Grant Types
 
-The [OAuth 2.0 Authorization Framework specification](https://tools.ietf.org/html/rfc6749) defines four grant types.
+The [OAuth 2.0 Authorization Framework specification](https://tools.ietf.org/html/rfc6749) defines four flows to get an access token. These flows are called **grant types**. Deciding which one is suited for your case depends mostly on the type of your client.
+
+- [Authorization Code](/api-auth/grant/authorization-code): used by Web Apps executing on a server. This is also used by mobile apps, using the [Proof Key for Code Exchange (PKCE) technique](/api-auth/grant/authorization-code-pkce).
+
+- [Implicit](/api-auth/grant/implicit): used by JavaScript-centric apps (Single Page Applications) executing on the user's browser.
+
+- [Resource Owner Password Credentials](/api-auth/grant/password): used by trusted apps.
 
-- **Authorization Code**: used by Web Apps executing on a server.
-- **Implicit**: used by JavaScript-centric apps (SPAs) executing on the user's browser.
-- **Resource Owner Password Credentials**: used by trusted apps.
-- **Client Credentials**: used for machine-to-machine communication.
+- [Client Credentials](/api-auth/grant/client-credentials): used for machine-to-machine communication.
 
 The specification also provides an extensibility mechanism for defining additional types.
 
 For details on how each grant type works and when it should be used refer to [API Authorization](/api-auth).
+
+
+## OAuth Endpoints
+
+OAuth 2.0 utilizes two endpoints: the **Authorization** endpoint and the **Token** endpoint.
+
+### Authorization Endpoint
```

Comment:
@cabralmartin I added two new sub-paragraphs since you reviewed this PR (how response type works, how response mode works). Do you want to have a look before I merge?